code/modules/cm_aliens/structures/special/pylon_core.dm

```diff
@@ -195,16 +194,24 @@
 	for(var/mob/living/carbon/xenomorph/xeno in hive_xenos)
 		if(!xeno.counts_for_slots)
 			hive_xenos -= xeno
-
 	var/real_total_xeno_count = length(hive_xenos) + linked_hive.stored_larva
 
-	if(real_total_xeno_count > (length(GLOB.alive_human_list) * ENDGAME_LARVA_CAP_MULTIPLIER))
+	var/groundside_humans_weighted_count = 0
+	for(var/mob/living/carbon/human/current_human as anything in GLOB.alive_human_list)
+		if(!(isspecieshuman(current_human) || isspeciessynth(current_human)))
+			continue
+		var/datum/job/job = GLOB.RoleAuthority.roles_for_mode[current_human.job]
+		if(!job)
+			continue
+		var/turf/turf = get_turf(current_human)
+		if(!is_ground_level(turf?.z))
+			groundside_humans_weighted_count += GLOB.RoleAuthority.calculate_role_weight(job)
+	if(real_total_xeno_count > (groundside_humans_weighted_count * ENDGAME_LARVA_CAP_MULTIPLIER))
 		return
```

Comment:
Since you cited "Current implementation encourages xenos to delay hijack as long as possible to get the maximum larva" you'd have more effect if hitting this cap in some way was communicated to the hive. I'm not really suggesting it should have a text announcement (if it does it should only do it infrequently and not be spammy); but likely would be good to think of maybe a way for this to be indicated in the hive status.

Reply:
I added messages, but I am afraid it still might be spammy. Especially because pylons provide larvae separately, so time between procs may differ significantly. Any better ideas?

Reply:
I probably can add some kind of timer that will check if hit_larva_limit is still FALSE after few minutes and send the message only then... Seems kinda lame, but cannot think of a better solution

Reply:
and still there can be a situation, when I send the message and the hit_larva instantly becomes true and sends another message =\\\

Reply:
It's almost like I need to use one timer for both messages and queue them correctly, hmmm

Reply:
As I mentioned, I think it would be better to be information in the hive_status rather than messages:
> I'm not really suggesting it should have a text announcement (if it does it should only do it infrequently and not be spammy); but likely would be good to think of maybe a way for this to be indicated in the hive status.

Reply:
E.g. here: 
![image](https://github.com/cmss13-devs/cmss13/assets/76988376/202ed7b3-3bae-44b6-984d-50764d7fc3e4)

Reply:
yeah I can use tgui I code via VM. I don't think anyone would check hivestatus to know pylon status though

Reply:
Almost every xeno uses this UI. And if theres new text near burrowed larva they will see it, because people look at that number all the time.

Reply:
Alright I figure something out a bit later